src/xunit.performance.api/XunitPerformanceHarnessOptions.cs

```diff
@@ -39,7 +40,46 @@ public string OutputDirectory
                 _outputDirectory = Path.IsPathRooted(value) ? value : Path.GetFullPath(value);
                 if (!Directory.Exists(_outputDirectory))
                 {
-                    Directory.CreateDirectory(_outputDirectory);
+                    try 
+                    {
+                        Directory.CreateDirectory(_temporaryDirectory);
+                    } catch (System.Exception) 
+                    {
+                        Console.Error.WriteLine("Couldn't create Directory " + _temporaryDirectory);
+                        throw;
+                    }
+                }
+            }
+        }
+
+        [Option("perf:tmpdir", Required = false, HelpText = "Specifies a writable directory to store temporary files.")]
+        public string TemporaryDirectory
+        {
+            get { return _temporaryDirectory; }
+
+            set
+            {
+                if (string.IsNullOrWhiteSpace(value))
+                {
+                    throw new Exception("The temporary directory name cannot be null, empty or white space.");
+                }
+
+                if (value.Any(c => Path.GetInvalidPathChars().Contains(c)))
+                {
+                    throw new Exception("Specified temporary directory name contains invalid path characters.");
+                }
+
+                _temporaryDirectory = Path.IsPathRooted(value) ? value : Path.GetFullPath(value);
+                if (!Directory.Exists(_temporaryDirectory))
```

Comment:
You actually don't need to test for this condition: `Directory.CreateDirectory(path)` already does this internally, and returns as a no-op if `path` already exists as a directory. See <https://docs.microsoft.com/en-us/dotnet/core/api/system.io.directory>.